lib/src/main/java/graphql/nadel/engine/transform/hydration/NadelHydrationTransform.kt

```diff
@@ -187,88 +273,99 @@ internal class NadelHydrationTransform(
-            return emptyList()
+            return NadelPreparedHydration {
+                emptyList()
+            }
         }
 
         val instruction = getHydrationFieldInstruction(state, instructions, executionContext.hooks, parentNode)
-            ?: return listOf(
-                NadelResultInstruction.Set(
-                    subject = parentNode,
-                    key = NadelResultKey(state.hydratedField.resultKey),
-                    newValue = null,
-                ),
-            )
+            ?: return NadelPreparedHydration {
+                listOf(
+                    NadelResultInstruction.Set(
+                        subject = parentNode,
+                        key = NadelResultKey(state.hydratedField.resultKey),
+                        newValue = null,
+                    ),
+                )
+            }
 
-        val actorQueryResults = coroutineScope {
-            NadelHydrationFieldsBuilder.makeActorQueries(
-                instruction = instruction,
-                aliasHelper = state.aliasHelper,
-                fieldToHydrate = fieldToHydrate,
-                parentNode = parentNode,
-                executionBlueprint = executionBlueprint,
-            ).map { actorQuery ->
-                async {
-                    val hydrationSourceService = executionBlueprint.getServiceOwning(instruction.location)!!
-                    val hydrationActorField =
-                        FieldCoordinates.coordinates(instruction.actorFieldContainer, instruction.actorFieldDef)
-                    val serviceHydrationDetails = ServiceExecutionHydrationDetails(
-                        timeout = instruction.timeout,
-                        batchSize = 1,
-                        hydrationSourceService = hydrationSourceService,
-                        hydrationSourceField = instruction.location,
-                        hydrationActorField = hydrationActorField,
-                        fieldPath = fieldToHydrate.listOfResultKeys
-                    )
-                    engine.executeTopLevelField(
-                        service = instruction.actorService,
-                        topLevelField = actorQuery,
-                        executionContext = executionContext,
-                        serviceHydrationDetails = serviceHydrationDetails,
-                    )
-                }
-            }.awaitAll()
-        }
+        val actorQueries = NadelHydrationFieldsBuilder.makeActorQueries(
+            instruction = instruction,
+            aliasHelper = state.aliasHelper,
+            fieldToHydrate = fieldToHydrate,
+            parentNode = parentNode,
+            executionBlueprint = executionBlueprint,
+        )
 
-        when (instruction.hydrationStrategy) {
-            is NadelHydrationStrategy.OneToOne -> {
-                // Should not have more than one query for one to one
-                val result = actorQueryResults.emptyOrSingle()
+        return NadelPreparedHydration {
```

Comment:
Code inside here should be the same, it's just wrapped in a callback now, instead of being executed immediately.